portal/portal-impl/impl/src/java/org/sakaiproject/portal/charon/SkinnableCharonPortal.java

```diff
@@ -1084,6 +1084,15 @@ public PortalRenderContext startPageContext(String siteType, String title,
 		rcontext.put("userSiteRole", role != null ? role.getId() : "");
 		rcontext.put("editorType", editorType);
 
+                Session session = SessionManager.getCurrentSession();
+                boolean justLoggedIn = false;
+                Object jli = session.getAttribute(Session.JUST_LOGGED_IN);
+                if (jli != null) {
+			justLoggedIn = true;
+			session.removeAttribute(Session.JUST_LOGGED_IN);
+		}
```

Comment:
```suggestion
                boolean justLoggedIn = session.getAttribute(Session.JUST_LOGGED_IN) != null;
                
               if (justLoggedIn) {
                        session.removeAttribute(Session.JUST_LOGGED_IN);
               }

```

Reply:
@kunaljaykam - Thanks, Kunal. I'll force push this revision soon.